### PR TITLE
Features and fixes needed for contract deployment test

### PR DESCRIPTION
**Pull request by steezeburger**
* add metro host and port to cli flags
* add and remove txs from local mempool in addition to metro, so that things like forge still work

### CODE REVIEW

File: grpc/execution/server.go [hunk 1]
```diff
@@ -92,6 +92,13 @@ func (s *ExecutionServiceServer) DoBlock(ctx context.Context, req *executionv1.D
 		return nil, fmt.Errorf("failed to insert block into blockchain (n=%d)", n)
 	}
 
+	// remove txs from original mempool
+	for _, block := range blocks {
+		for _, tx := range block.Transactions() {
+			s.eth.TxPool().RemoveTx(tx.Hash())
+		}
+	}
```
Comment (noot) [inline grpc/execution/server.go]:
```suggestion
	for _, tx := range block {
		s.eth.TxPool().RemoveTx(tx.Hash())
	}
```

can be simplified as there's only one block

File: internal/ethapi/api.go [hunk 2]
```diff
@@ -1694,12 +1694,19 @@ func SubmitTransaction(ctx context.Context, b Backend, tx *types.Transaction) (c
...
+
 	// send to metro instead of eth mempool
 	txBytes, err := tx.MarshalBinary()
 	if err != nil {
 		return common.Hash{}, err
 	}
-	if err := submitMetroTransaction(txBytes); err != nil {
+
+	metro_api := NewMetroAPI(b.MetroGRPCEndpoint())
+	if err := metro_api.SubmitTransaction(txBytes); err != nil {
```
Comment (noot) [inline internal/ethapi/api.go]:
```suggestion
	metroAPI := NewMetroAPI(b.MetroGRPCEndpoint())
	if err := metroAPI.SubmitTransaction(txBytes); err != nil {
```

golang naming nit

File: node/defaults.go [hunk 3]
```diff
@@ -36,9 +36,14 @@ const (
...
 	DefaultAuthHost    = "localhost" // Default host interface for the authenticated apis
 	DefaultAuthPort    = 8551        // Default port for the authenticated apis
+
 	// grpc
 	DefaultGRPCHost = "[::1]" // Default host interface for the gRPC server
 	DefaultGRPCPort = 50051   // Default port for the gRPC server
+
+	// metro
+	DefaultMetroGRPCHost = "127.0.0.1" // Default host interface for the metro gRPC server
+	DefaultMetroGRPCPort = 9090        // Default port for the metro gRPC server
```
Comment (noot) [inline node/defaults.go]: are these used? don't see them used anywhere

File: grpc/execution/server.go [hunk 4]
```diff
@@ -92,6 +92,11 @@ func (s *ExecutionServiceServer) DoBlock(ctx context.Context, req *executionv1.D
 		return nil, fmt.Errorf("failed to insert block into blockchain (n=%d)", n)
 	}
 
+	// remove txs from original mempool
+	for _, tx := range block.Transactions() {
```
Comment (joroshiba) [inline grpc/execution/server.go:96]: Should we be removing all transactions in the request even if they don't make the block, or maybe doing something with transactions that don't make it into the block?

Comment (noot) [reply grpc/execution/server.go:96]: all the transactions in here did make it into the block, so this should be fine

**Review by noot — APPROVED** [review]

File: grpc/execution/server.go (hunk 4 above)
Comment (joroshiba) [reply grpc/execution/server.go:96]: We have filtering to stop transactions that can't be executed from being included. Those txs will NEVER be in a block, should they remain in the mempool?

Edit: I think this is good for now but something to think on.

**Review by joroshiba — APPROVED** [review]

**Pull request merged**